src/Model/Common/StatusInformation.php

```diff
@@ -42,7 +43,7 @@ class StatusInformation
      */
     public function getStatusCode(): int
     {
-        return $this->statusCode;
+        return intval($this->statusCode);
```

Comment:
[`intval` is discouraged](https://github.com/magento/magento-coding-standard/blob/v5/Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php#L220), use `(int) $value` instead

Reply:
Will that convert a string containing digits to an int (the issue to be solved)?
I'm not yet fluent in the new typesafety features and can't find an answer online.

Reply:
Will that convert a string containing digits to an int (the issue to be solved)?
I'm not yet fluent in the new typesafety features and can't find an answer online.

Reply:
[Type casting](https://www.php.net/manual/en/language.types.type-juggling.php#language.types.typecasting) exists for ages, it is not a new feature. Both statements are interchangeable here but the former is discouraged by Magento coding standard in favour of the latter. As I cannot replicate the issue, you may validate yourself that it still solves the issue.

Reply:
We'll test it in the real world and update the pull request accordingly. No problem.

Reply:
tested, works. Code updated.